datatypes/src/primitives/time_step.rs

```diff
@@ -32,7 +32,7 @@ pub enum TimeGranularity {
 #[cfg_attr(feature = "postgres", derive(ToSql, FromSql))]
 pub struct TimeStep {
     pub granularity: TimeGranularity,
-    pub step: u32,
+    pub step: u32, // TODO: ensure on deserialization it is > 0
```

Comment:
What about [NonZeroU32](https://doc.rust-lang.org/std/num/struct.NonZeroU32.html)?

Reply:
good idea but needs to be a separate task. The TimeStep is used in a way where the step can be 0. 
* There are tests that assure it works with 0
* the TimeStepIter multiples the step with a running index that starts at zero
* the num_step function uses steps with 0 value

there are many changes across the project necessary which are beyond the scope of this PR

Reply:
add a task / idea?

Reply:
https://trello.com/c/3a9SrZnQ